src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2022 Daniel Roca González. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Roca González
+-/
+import analysis.inner_product_space.projection
+import analysis.inner_product_space.dual
+import analysis.normed_space.banach
+import topology.metric_space.antilipschitz
+
+/-!
+# The Lax-Milgram Theorem
+
+
+We consider an Hilbert space `V` over `ℝ`
+equipped with a bounded bilinear form `B : V →L[ℝ] V →L[ℝ] ℝ`.
+We define a property `is_coercive` for `B : V →L[ℝ] V →L[ℝ] ℝ`,
+such that `is_coercive B` iff
+`∃ C, (0 < C) ∧ ∀ u, C * ∥u∥ * ∥u∥ ≤ B u u`.
+
+Under the hypothesis that `B` is coercive
+we prove the Lax-Milgram theorem:
+that is, the map `continuous_linear_map_of_bilin` from `analysis.inner_product_space.dual`
+can be upgraded to a continuous equivalence `lax_milgram_equiv : V ≃L[ℝ] V`.
+
+## References
+
+* We follow the notes of Peter Howard's Spring 2020 *M612: Partial Differential Equations* lecture,
+  see[howard]
+
+## Tags
+
+dual, Lax-Milgram
+-/
+
+noncomputable theory
+open is_R_or_C linear_map continuous_linear_map inner_product_space
+open_locale real_inner_product_space nnreal
+
+universe u
+
+/--
+A bounded bilinear form in an inner product space is *coercive*
+if there is some positive constant C such that `C * ∥u∥ * ∥u∥ ≤ B u u`.
+-/
+def is_coercive
+  {V : Type u} [inner_product_space ℝ V] (B : V →L[ℝ] V →L[ℝ] ℝ) : Prop :=
+∃ C, (0 < C) ∧ ∀ u, C * ∥u∥ * ∥u∥ ≤ B u u
+
+namespace is_coercive
+variables {V : Type u} [inner_product_space ℝ V] [complete_space V]
+variables {B : V →L[ℝ] V →L[ℝ] ℝ}
+
+local postfix `♯`:1025 := @continuous_linear_map_of_bilin ℝ V _ _ _
+
+lemma bounded_below (coercive : is_coercive B) :
+  ∃ C, 0 < C ∧ ∀ v, C * ∥v∥ ≤ ∥B♯ v∥ :=
+begin
+  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
+  refine ⟨C, C_ge_0, _⟩,
+  intro v,
+  by_cases h : 0 < ∥v∥,
+  { refine (mul_le_mul_right h).mp _,
+    exact calc C * ∥v∥ * ∥v∥
+        ≤ B v v : coercivity v
+    ... = ⟪B♯ v, v⟫_ℝ : by simp
+    ... ≤ ∥B♯ v∥ * ∥v∥ : real_inner_le_norm (B♯ v) v, },
+  { have : v = 0 := by simpa using h,
+    simp [this], }
+end
+
+lemma antilipschitz_of_lax_milgram (coercive : is_coercive B) :
```

Comment:
I think you can just name this lemma `antilipschitz`, since it's now in the `is_coercive` namespace so an "of `is_coercive`" is somehow implied.

Reply:
You are correct, this works.